addons/stock_account/models/product.py

```diff
@@ -111,31 +111,48 @@ def write(self, vals):
             self.filtered(lambda p: p.cost_method != 'fifo')._change_standard_price(vals['standard_price'])
         return super(ProductProduct, self).write(vals)
 
-    @api.depends('stock_valuation_layer_ids')
-    @api.depends_context('to_date', 'company')
-    def _compute_value_svl(self):
-        """Compute totals of multiple svl related values"""
-        company_id = self.env.company
-        self.company_currency_id = company_id.currency_id
+    def _get_valuation_layer_group_domain(self):
+        company_id = self.env.company.id
         domain = [
             ('product_id', 'in', self.ids),
-            ('company_id', '=', company_id.id),
+            ('company_id', '=', company_id),
         ]
         if self.env.context.get('to_date'):
             to_date = fields.Datetime.to_datetime(self.env.context['to_date'])
             domain.append(('create_date', '<=', to_date))
-        groups = self.env['stock.valuation.layer']._read_group(domain, ['value:sum', 'quantity:sum'], ['product_id'])
+        return domain
+
+    def _get_valuation_layer_group_fields(self):
+        return ['value:sum', 'quantity:sum']
+
+    def _get_valuation_layer_groups(self):
+        domain = self._get_valuation_layer_group_domain()
+        group_fields = self._get_valuation_layer_group_fields()
+        return self.env['stock.valuation.layer'].read_group(domain, group_fields, ['product_id'])
+
+    def _prepare_valuation_layer_field_values(self, group):
```

Comment:
I am not sure about the use of `group` here ... in the original code, without a separate method, the variable was called `group` because it was one of the grouped results I guess. In this case, however, maybe we could pass the relevant values separately? So: `(self, value_svl, quantity_svl)` or something similar?

Reply:
Oh, I didn't mean changing the `_get_valuation_layer_group_fields` method, but rather `_prepare_valuation_layer_field_values` method. I think that passing an argument called "group" is a bit unclear: what kind of group it it? I think it would be better to extract the individual values from the group in `_compute_value_svl` and pass them separately here. The `_get_valuation_layer_group_fields` can stay as it is :)

Reply:
I cannot pass the individual arguments there, otherwise `_prepare_valuation_layer_field_values` will not get all of the values. 

Our uses case is the following, we extending `_get_valuation_layer_group_fields` with new fields. And those fields will then be used in `_prepare_valuation_layer_field_values`. Thats why it makes no sense for us to pass just the individual (value, quantity) values to `_prepare_valuation_layer_field_values`. :-)

But a renaming of the var would be helpful.

Reply:
Sorry for the delay. I've read your comment and now I understand your use case. Let's keep a single argument in that case ;) Ideally, however, it would be renamed, maybe to `valuation_layer_group` or `grouped_valuation_layer_fields` or anything other than just `group` ? :)

I'll ask someone for a review, to confirm that the code is OK, and then an approval :+1:

Reply:
Yes lets rename the argument.